lib/parse.js

```diff
@@ -237,14 +237,36 @@ var normalizeParseOptions = function normalizeParseOptions(opts) {
     };
 };
 
+var decodeObject = function (obj, opts) {
+    if (opts.decoder === defaults.decoder) {
+        return obj;
+    }
+    if (typeof obj !== 'object') {
+        return obj;
+    }
+    var decodedObj = {};
```

Comment:
this needs to respect `plainObjects`.
```suggestion
    var decodedObj = opts.plainObjects ? Object.create(null) : {};
```